src/components/organisms/UiDatepicker/_internal/UiDatepickerYearInput.vue

```diff
@@ -61,9 +61,9 @@ const year = computed({
 });
 const validationError = computed(() => (year.value.length === 4 && !props.valid));
 const hasError = computed(() => (validationError.value || unfulfilledYearError.value || props.error));
-async function checkYear(event: InputEvent): Promise<void> {
+async function checkYear(event: Event) {
   unfulfilledYearError.value = false;
-  const inputValue = event.data;
+  const inputValue = (event as InputEvent).data;
```

Comment:
Q: Why these changes? Wasn't it cleaner with `InputEvent` in the function parameter and without this casting? And why did you remove the function's value returned type? 😅

Reply:
1. If we have `InputEvent` in the function parameter then we have an error in the template because the input payload has type `Event`
2. I removed it, because the typescript knows it by inference